onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -112,7 +112,14 @@ Status SequentialExecutor::Execute(const SessionState& session_state,
 
       kernel_begin_time = session_state.Profiler().StartTime();
     }
-    ORT_RETURN_IF_ERROR(p_op_kernel->Compute(&op_kernel_context));
+
+    const auto& compute_status = p_op_kernel->Compute(&op_kernel_context);
+    if (!compute_status.IsOK()) {
+      LOGS(logger, ERROR) << "Non-zero status code returned while running Node: ",
+                             p_op_kernel->Node().Name(), " Status Message: ", compute_status.ErrorMessage();
+      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Non-zero status code returned while running Node: ",
```

Comment:
Not sure if anything external depends on the failure category or code. You may want to maintain those instead of always overwriting with ONNXRUNTIME and FAIL.

Reply:
That makes sense, thanks, fixed.